tracee-ebpf/tracee/tracee.bpf.c

```diff
@@ -3447,5 +3750,156 @@ int BPF_KPROBE(trace_security_bpf_map)
     return 0;
 }
 
+static __always_inline bool skb_revalidate_data(struct __sk_buff *skb, uint8_t **head, uint8_t **tail, const u32 offset) {
+    if (*head + offset > *tail) {
+        if (bpf_skb_pull_data(skb, offset) < 0) {
+            return false;
+        }
+
+        *head = (uint8_t *)(long)skb->data;
+        *tail = (uint8_t *)(long)skb->data_end;
+
+        if (*head + offset > *tail) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+static __always_inline int tc_probe(struct __sk_buff *skb, bool ingress) {
```

Comment:
I'm here on my review. Have almost finished the golang part, finishing this and I'm ready to finish it.